libmscore/element.cpp

```diff
@@ -2187,24 +2346,40 @@ void Element::autoplaceSegmentElement(qreal minDistance)
                   qreal yd = d + minDistance;
                   if (placeAbove())
                         yd *= -1.0;
+                  if (offsetChanged()) {
+                        // user moved element within the skyline
+                        // we may need to adjust minDistance, yd, and/or offset
+                        bool inStaff = placeAbove() ? r.bottom() + rebase > 0.0 : r.top() + rebase < staff()->height();
```

Comment:
While the placement property might not be sufficient to determine position relative to the staff, it *is* how autoplace normally works - we rely on this many places.  In particular, it's what we use to determine which skyline to try to avoid, when calculating yd.  So I'm not sure this code needs to be adjusted because of that, but maybe you can find a specific case?

Accounting for height correctly is good indeed, I'll do that.

Reply:
Actually, regarding height - the layout code for below placement (for all elements I could find that honor it) also just uses plain staff()->height(), so elements placed below staff are already placed incorrectly if the height changes mid-score.  I see there is a TODO in Staff::height() to use tick, at some point someone would go in and deal with that.  But I'm inclined to leave my code here alone, then, for consistency, and to make it easier to update all Staff:height() calls together when the time comes.